{
  delayTestFinish(DELAY_TEST_FINISH);
  final RuntimeException exception1=new RuntimeException("first exception");
  final RuntimeException exception2=new RuntimeException("second exception");
  final CountingReceiver count=new CountingReceiver();
  SimpleFooRequest context=req.simpleFooRequest();
  context.pleaseCrash(42).to(count);
  context.returnNullString().to(new ThrowingReceiver<String>(exception1));
  context.returnNullString().to(count);
  fireContextAndCatch(context,new ThrowingReceiver<Void>(exception2),new GWT.UncaughtExceptionHandler(){
    public void onUncaughtException(    Throwable e){
      if (e instanceof UmbrellaException) {
        count.assertCounts(1,1,0);
        Set<Throwable> causes=((UmbrellaException)e).getCauses();
        assertEquals(2,causes.size());
        assertTrue(causes.contains(exception1));
        assertTrue(causes.contains(exception2));
        finishTestAndReset();
      }
 else {
        defaultUncaughtExceptionHandler.onUncaughtException(e);
      }
    }
  }
);
}
